### PR TITLE
Created BookingDao interface and BookingDaoImpl

### PR DESCRIPTION
**Pull request by bigno78**

### CODE REVIEW

**Review by Fuyunori — APPROVED** [review]

File: src/main/java/tennisclub/dao/BookingDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    @Override
+    public List<Booking> findByEndTime(LocalDateTime endTime) {
+        return em.createQuery("select b from Booking b where b.endTime = :endTime", Booking.class)
+                .setParameter("endTime", endTime)
+                .getResultList();
+    }
+
+    @Override
+    public List<Booking> findByTimeInterval(LocalDateTime from, LocalDateTime to) {
+        return em.createQuery("select b from Booking b where b.endTime >= :from and b.startTime <= :to", Booking.class)
```
Comment (paveltobias) [inline src/main/java/tennisclub/dao/BookingDaoImpl.java:54]: This is a matter of opinion/use-case, but would strict inequalities (`>` and `<`) not make more sense? E.g. if an event **ends** at 16:00, I would not expect to see it in a list of events happening **from** 16:00 on.

File: src/main/java/tennisclub/dao/BookingDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    @Override
+    public Booking update(Booking booking) {
+        return em.merge(booking);
+    }
+
+    @Override
+    public void delete(Booking booking) {
+        if (!em.contains(booking)) {
+            em.merge(booking);
+        }
```
Comment (paveltobias) [inline src/main/java/tennisclub/dao/BookingDaoImpl.java:74]: Good thinking! Deleting detached entities without the need to re-find them or "`update()`" them first is a cool feature! Let me steal this later. :smile:

Comment (bigno78) [reply src/main/java/tennisclub/dao/BookingDaoImpl.java:74]: I already stole this from @Fuyunori. :smile:

File: src/main/java/tennisclub/dao/BookingDaoImpl.java (hunk 1 above)
Comment (bigno78) [reply src/main/java/tennisclub/dao/BookingDaoImpl.java:54]: Hmmm, yes I think you are right. I will change it here and in EventDao as well.

**Review by paveltobias — APPROVED** [review]

**Pull request merged**